venv/lib/python3.9/site-packages/pip/_internal/network/auth.py

```
@@ -0,0 +1,312 @@
+)
+from pip._internal.vcs.versioncontrol import AuthInfo
+
+logger = logging.getLogger(__name__)
+
+Credentials = Tuple[str, str, str]
+
+try:
+    import keyring
+except ImportError:
+    keyring = None
+except Exception as exc:
```

Ruff (0.14.8)

Do not catch blind exception: `Exception`

(BLE001)